ccube.nim

```diff
@@ -66,142 +66,101 @@ Examples:
+      of "noloop":
         try:
-          filePath = param.split(':')[1]      
-        except IndexDefect:
-          stderr.write("Error:Bad use of --file switch,valid file path expected") 
+          result.noLoopSwitch = parseBool(val)
+        except ValueError:
+          stderr.write(fmt"Error: '{val}' is not a valid boolean value.")
           quit()
-    
-    if len(filePath) > 0:
-      discard
-    else:
-      filePath = fmt"{getCurrentDir()}{DirSep}iterationData.txt"
-    
-    if len(number) > 0:
-      discard
-    else:
-      number = $rand(1..1000)
-    
-    if len(noLoopSwitchBool) > 0:
-      discard
-    else:
-      noLoopSwitchBool = "false"
-    
-    result = (noLoopSwitchBool,number,filePath)
-  
-  else:
-    showUsage()
+      of "n":
+        for c in val:
+          if c notin Digits:
+            stderr.write(fmt"Error: '{val}' is not a valid number.")
+            quit()
+        result.number = val
+      of "file":
+        result.filePath = val
+    else: discard
 
-proc isEven(number:BigInt):bool =
-  if number mod 2 == 0:
-    result = true
-  else:
-    result = false
+  if result.filePath.len == 0:
+    result.filePath = getCurrentDir() / "iterationData.txt"
 
-proc readJson():(string,string,string) =
-  var isFile = fileExists("ccube_conf.json")
+  if result.number.len == 0:
+    result.number = $rand(1..1000)
+
+proc isEven(number: BigInt): bool =
+  result = number mod 2 == 0
+
+proc readJson(): (ProgramArgs, bool) =
+  var isFile = fileExists("ccube_config.json")
   if isFile:
     try:
-      let
-        readJsonData = readFile("ccube_conf.json")
-        jsonNode = parseJson(readJsonData)
-        noLoopSwitchBool = $jsonNode["noLoopSwitch"]
-        number = $jsonNode["number"]
-        filePath = $jsonNode["filePath"]
-      result = (noLoopSwitchBool,number,filePath)
+      let args = parseFile("ccube_config.json").to(ProgramArgs)
+      result = (args, true)
     except KeyError:
-      stderr.write("Error:An error occured while reading json keys,Invalid Keys!\n")
-      result = ("-1","-1","-1")
-  else:
-    result = ("-1","-1","-1")
+      stderr.write("An error occured while reading JSON configuration.")
 
-
-proc main(cmdArgs:(string,string,string)):void = 
+proc doWork(args: ProgramArgs) =
   var
-    noLoopSwitchBool = cmdArgs[0].strip(chars = {'"'})   # strip " from string if exists
-    number = initBigInt(cmdArgs[1].strip(chars = {'"'}))
-    filePath = cmdArgs[2].strip(chars = {'"'})
+    number = initBigInt(args.number)
     isLooping = true
-    incrementer:BigInt
-    noOfIterations:BigInt = initBigInt("0")
-  while isLooping or noLoopSwitchBool == "false":
+    incrementer: BigInt
+    noOfIterations = initBigInt("0")
+  
+  while isLooping:
     incrementer = number
     while number != 1:
       if isEven(number):
         number = number div 2
-        inc noOfIterations
       else:
         number = (number * 3) + 1
-        inc noOfIterations
+      inc noOfIterations
```

Comment:
Since you call the same `inc noOfIterations` in both cases, it's better to just move it out of the if statement